daemon/src/db.rs

```diff
@@ -161,21 +188,163 @@ pub async fn append_event(
             (select id from cfds where cfds.uuid = $1),
             $2, $3, $4
         )"##,
-    )
-    .bind(&event.id)
-    .bind(&event_name)
-    .bind(&event_data)
-    .bind(&event.timestamp)
-    .execute(conn)
-    .await?;
+        )
+        .bind(&event.id)
+        .bind(&event_name)
+        .bind(&event_data)
+        .bind(&event.timestamp)
+        .execute(&mut conn)
+        .await?;
+
+        if query_result.rows_affected() != 1 {
+            anyhow::bail!("failed to insert event");
+        }
+
+        tracing::info!(event = %event_name, order_id = %event.id, "Appended event to database");
 
-    if query_result.rows_affected() != 1 {
-        anyhow::bail!("failed to insert event");
+        Ok(())
     }
 
-    tracing::info!(event = %event_name, order_id = %event.id, "Appended event to database");
+    /// Load a CFD in its latest version from the database.
+    pub async fn load_cfd<C>(&self, id: OrderId, args: C::CtorArgs) -> Result<C>
+    where
+        C: CfdAggregate,
+    {
+        let mut conn = self.inner.acquire().await?;
 
-    Ok(())
+        let cache_key = (TypeId::of::<C>(), id);
+
+        let cfd = match self.aggregate_cache.remove(&cache_key).await {
```

Comment:
does the version always change between this function invocations? Seems like it would be possible to remove and re-insert the same key if version didn't change?

Furthermore, wouldn't it be preferable to use `get()` here, as that would allow concurrent access without possibility of a cache miss?

Reply:
To apply more events I need ownership of the CFD struct so I need to remove it!

Reply:
an alternative would be probably doing a `clone()` then and only updating later on. 
With the clone() we'd lose some of the benefit of trying to make it faster in the first place. though. 

probably a bit cleaner for me, but I can see how that could be subjective.